presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java

```diff
@@ -1101,7 +1104,12 @@ public void moveToCurrentRow()
     public Statement getStatement()
             throws SQLException
     {
-        throw new NotImplementedException("ResultSet", "getStatement");
+        if (statement.isPresent()) {
+            return statement.get();
+        }
+        else {
+            throw new NotImplementedException("ResultSet", "getStatement");
```

Comment:
The method _is_ implemented, but the context doesn't allow us to provide a Statement object
(eg some DatabaseMetaData method returning a ResultSet).

```suggestion
            throw new SQLException("Statement not available");
```